include/compat/sys/stat.h

```diff
@@ -118,4 +118,12 @@
 
 #endif
 
+#ifdef _WIN32
+int posix_fstat(int fd, struct stat *statbuf);
+
+#ifndef NO_REDEF_POSIX_FUNCTIONS
+#define fstat(fd, statbuf) posix_fstat(fd, statbuf)
```

Comment:
This is fine for now, but I think we should consider switching things to a `libressl_*` prefix also for windows. Similarly as here:

https://github.com/libressl/portable/blob/73779a46bf49c4f53cc4b81993135a7408a01963/include/compat/stdio.h#L21-L39